File: LeetCode/Easy/0771-jewels-and-stones/0771-jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        int[] counter = new int[58];

        for (char c : stones.toCharArray()) {
            counter[c - 'A']++;
        }

        int answer = 0;

        for (char c : jewels.toCharArray()) {
            answer += counter[c - 'A'];
        }

        return answer;
    }
}

File: LeetCode/Easy/0771-jewels-and-stones/0771-jewels-and-stones.kt
class Solution {
    fun numJewelsInStones(jewels: String, stones: String): Int {
        val jSet = HashSet<Char>()

        for (j in jewels) {
            jSet.add(j)
        }

        var answer = 0

        for (s in stones) {
            if (jSet.contains(s)) {
                answer++
            }
        }

        return answer
    }
}